lib/authenticate.js

```diff
@@ -56,5 +113,57 @@ const authenticate = url =>
       opn(cliLoginUrl, { wait: false })
     })
   })
+}
+
+async function authenticate (url) {
+  const mockJWT = process.env.CLINIC_JWT
+  const mockFail = process.env.CLINIC_MOCK_AUTH_FAIL
+  if (mockJWT) {
+    await saveToken(url, mockJWT)
+    return mockJWT
+  }
+  if (mockFail) {
+    throw new Error('Auth artificially failed')
+  }
+
+  // Use cached token if it's not expired
+  try {
+    const existingJWT = await loadToken(url)
+    if (existingJWT && validateToken(existingJWT)) {
+      return existingJWT
+    }
+  } catch (err) {}
```

Comment:
should we log anything in this case?

Reply:
entering this catch means the token file could not be read, which would normally mean that there is no ~/.node-clinic-rc file yet, eg when `clinic upload` is first used. that is a little weird though, i'll rework this a bit